### PR TITLE
Airport Challenge - Benedetta Arinci

### PR DESCRIPTION
**Pull request by BeneArinci**
- The code is satisfying all 6 user stories and some edge cases
- What I've done is complete but there is a huge room for imrpovement
- Test coverage is 100% and all the tests are working

### CODE REVIEW

File: lib/airport.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+require './lib/plane.rb'
+class Airport
+  def initialize(capacity = 20)
+    @planes_in = []
+    @capacity = capacity
+  end
+
+  def show_planes_in
```
Comment (Katie-McDonagh) [inline lib/airport.rb:8]: Great naming of methods, easy to understand and follow

File: spec/airport_spec.rb [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    expect(subject.instance_variable_get :@planes_in)
+  end
+  it 'creates a capacity instance variable' do
+    expect(subject.instance_variable_get :@capacity).to eq(20)
+  end
+  it 'gives possibility to change the max capacity when the new airport is init' do
+    expect(Airport.new(30).instance_variable_get :@capacity).to eq(30)
+  end
+
+  context 'landing' do
```
Comment (Katie-McDonagh) [inline spec/airport_spec.rb:18]: Great use of 'context' RSpec syntax, grouping together tests makes the file easy to understand and follow

File: spec/airport_spec.rb [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      p2 = Plane.new
+      subject.land(p1)
+      subject.land(p2)
+      expect(subject.show_planes_in).to match_array([p1, p2])
+    end
+  end
+
+end
+
+describe Plane do
```
Comment (Katie-McDonagh) [inline spec/airport_spec.rb:95]: suggestion - create a separate spec file for Plane Class, I'm not sure why but it seems to be convention to have a separate spec file for each Class like they do in this tutorial : https://code.tutsplus.com/tutorials/ruby-for-newbies-testing-with-rspec--net-21297. I think the general rule is if the class has its own file like 'plane.rb' does it should have a corresponding plane_spec.rb file to run tests in.

File: spec/airport_spec.rb [hunk 4]
```diff
@@ -0,0 +1,99 @@
+require 'airport'
+require 'plane'
+
+describe Airport do
+  it 'creates an instance of the class Airport' do
```
Comment (Katie-McDonagh) [inline spec/airport_spec.rb:5]: suggestion - perhaps the test could be refactored slightly by using rspec one liners: https://relishapp.com/rspec/rspec-core/docs/subject/one-liner-syntax

File: README.md [hunk 5]
```diff
@@ -13,78 +13,19 @@ Airport Challenge
...
-
-* **Submit a pull request early.**
+Areas of improvement
+-----
 
-* Finally, please submit a pull request before Monday at 9am with your solution or partial solution.  However much or little amount of code you wrote please please please submit a pull request before Monday at 9am.
+I want to keep on working on this software because there are several areas I would like to improve. In particular:
+* forecast method shouldn't be part of the main class Airport but it should be part of a specific Class Weather;
+* Airport class should be able to allow multiple planes to land and take_off at the same time
+* I am sure there are other 10000 useful changes I might do so... stay tuned! :)
```
Comment (Katie-McDonagh) [inline README.md:31]: Great that you mention things to improve in your README file, perhaps it could be useful for someone reading the project, to see the user stories you used, and to mention the object/classes you made, and the methods you used
